lib/src/components/NxModal/__tests__/NxModal.test.tsx

```diff
@@ -290,4 +291,59 @@ describe('NxModal', function() {
       expect(buttons[0]).toHaveFocus();
     });
   });
+
+  it('should cycle through focusable elements inside stacked modals', async function() {
+    const user = userEvent.setup();
+
+    const Fixture = () => {
+      const [showModal, toggleModal] = useToggle(false);
+      const [showModal2, toggleModal2] = useToggle(false);
+      return (
+        <>
+          <button data-testid="open-modal-1" onClick={toggleModal}>Open</button>
+          { showModal &&
+            <NxModal onClose={toggleModal}>
+              <button data-testid="tabbable-modal-1">Modal1</button>
+              <button data-testid="tabbable-modal-1">Modal1</button>
+              <button data-testid="open-modal-2" onClick={toggleModal2}>Open</button>
+            </NxModal>
+          }
+          { showModal2 &&
+            <NxModal onClose={toggleModal2}>
+              <button data-testid="tabbable-modal-2">Modal2</button>
+              <button data-testid="tabbable-modal-2">Modal2</button>
+            </NxModal>
+          }
+        </>
+      );
+    };
+
+    const { getByTestId, getAllByTestId } = render(<Fixture />);
+    const openModal1Button = getByTestId('open-modal-1');
+
+    await user.click(openModal1Button);
+    await runTimers();
+
+    const modal1Tabbables = getAllByTestId('tabbable-modal-1');
+    const openModal2Button = getByTestId('open-modal-2');
+
+    expect(modal1Tabbables[0]).toHaveFocus();
+    await user.tab();
+    expect(modal1Tabbables[1]).toHaveFocus();
+    await user.tab();
+    expect(openModal2Button).toHaveFocus();
+    await user.tab();
+    expect(modal1Tabbables[0]).toHaveFocus();
+
+    await user.click(openModal2Button);
+    await runTimers();
+
+    // should cycle tabbable elements in modal 2 properly
+    const modal2Tabbables = getAllByTestId('tabbable-modal-2');
+    expect(modal2Tabbables[0]).toHaveFocus();
+    await user.tab();
+    expect(modal2Tabbables[1]).toHaveFocus();
+    await user.tab();
+    expect(modal2Tabbables[0]).toHaveFocus();
+  });
 });
```

Comment:
Need a test for the scenario described in https://github.com/sonatype/sonatype-react-shared-components/pull/1037/files#r1154650216

Reply:
> Need a test for the scenario described in https://github.com/sonatype/sonatype-react-shared-components/pull/1037/files#r1154650216

Added that test and simplified AbstractDialog further here: 5846ccae98a610cc71ff31f4265d6e958d143b65

Reply:
Looks like I'll have to write a function test to test the form one (display: none) doesn't behave nicely in jest-dom